examples/context7-rs/src/lib.rs

```diff
@@ -0,0 +1,358 @@
+                    }
+                } else {
+                    DocsResponse {
+                        success: true,
+                        content: Some(body),
+                        error: None,
+                    }
+                }
+            }
+            Err(message) => DocsResponse {
+                success: false,
+                content: None,
+                error: Some(message),
+            },
+        }
+    }
+}
+
+fn make_request(
+    method: Method,
+    endpoint: &str,
+    body: Option<&str>,
+    api_key: Option<&str>,
+) -> Result<String, String> {
+    let headers = create_headers(api_key, body.is_some())?;
+    let request = OutgoingRequest::new(headers);
+
+    request
+        .set_method(&method)
+        .map_err(|_| "Failed to set HTTP method".to_string())?;
+    request
+        .set_scheme(Some(&Scheme::Https))
+        .map_err(|_| "Failed to set scheme".to_string())?;
+    request
+        .set_authority(Some(CONTEXT7_HOST))
+        .map_err(|_| "Failed to set authority".to_string())?;
+
+    let mut path = String::with_capacity(BASE_PATH.len() + endpoint.len() + 1);
+    path.push_str(BASE_PATH);
+    if endpoint.starts_with('/') {
+        path.push_str(endpoint);
+    } else {
+        path.push('/');
+        path.push_str(endpoint);
+    }
+
+    request
+        .set_path_with_query(Some(&path))
+        .map_err(|_| "Failed to set path".to_string())?;
+
+    if let Some(payload) = body {
+        let outgoing_body = request
+            .body()
+            .map_err(|_| "Failed to acquire request body".to_string())?;
+        write_body(outgoing_body, payload.as_bytes())?;
+    }
+
+    let future = outgoing_handler::handle(request, None)
+        .map_err(|error| format!("HTTP request failed: {error:?}"))?;
+    let response = wait_for_response(&future)?;
+    let status = response.status();
+
+    let body = read_response_body(response)?;
+    if (200..300).contains(&status) {
+        Ok(body)
+    } else {
+        Err(if body.trim().is_empty() {
+            format!("HTTP {status}")
+        } else {
+            format!("HTTP {status}: {}", body)
+        })
+    }
+}
+
+fn create_headers(api_key: Option<&str>, has_body: bool) -> Result<Fields, String> {
+    let headers = Fields::new();
+    append_header(&headers, "user-agent", DEFAULT_USER_AGENT)?;
+    append_header(&headers, "accept", "*/*")?;
+
+    if has_body {
+        append_header(&headers, "content-type", "application/json")?;
+    }
+
+    if let Some(key) = api_key.and_then(|value| {
+        let trimmed = value.trim();
+        if trimmed.is_empty() {
+            None
+        } else {
+            Some(trimmed)
+        }
+    }) {
+        append_header(&headers, "context7-api-key", key)?;
+    }
+
+    Ok(headers)
+}
+
+fn append_header(headers: &Fields, key: &str, value: &str) -> Result<(), String> {
+    let normalized_key = key.to_ascii_lowercase();
+    let values = vec![value.as_bytes().to_vec()];
```

Comment:
Super nit: You don't need to alloc a vec here, `[value.as_bytes().to_vec()]` works just fine